src/common/base/src/base/runtime_tracker.rs

```diff
@@ -85,24 +83,27 @@ impl<'a> Drop for Entered<'a> {
     }
 }
 
-/// A guard that resets the `PANICKING` flag when dropped.
-pub(crate) struct Panicking;
+/// A guard that resets the `UNLIMITED_FLAG` flag when dropped.
+pub struct UnlimitedMemGuard {
+    saved: bool,
+}
 
-impl Panicking {
+impl UnlimitedMemGuard {
     #[must_use]
-    pub(crate) fn enter_panicking() -> Self {
-        PANICKING.store(true, Ordering::Relaxed);
-        Self
+    pub(crate) fn enter_unlimited() -> Self {
+        let saved = UNLIMITED_FLAG.load(Ordering::Relaxed);
+        UNLIMITED_FLAG.store(true, Ordering::Relaxed);
+        Self { saved }
```

Comment:
I'm confused. 🤔  This method is not used?

Reply:
yes. we will use it in the future. this will be used for some global services to run in unlimited memory.